lib/mirage/mirage.mli

```diff
@@ -15,19 +15,103 @@
+      we have defined {e a set of OCaml module types} that encode these
+      conventions in a statically enforcable way. We make no compatibility
+      guarantees at the C level, but compile those on a best-effort basis.
+
+    - Finally, MirageOS is also a {e metaprogramming compiler} that generates
+      OCaml code. It takes as input: the OCaml source code of a program and all
+      of its dependencies, the full description of the deployment target,
+      including configuration values (like the HTTP port to listen on, or the
+      private key or the service being deployed). The `mirage`CLI tool uses all
+      of these to {e generate a executable unikernel}: a specialised binary
+      artefact containing only the code what is needed to run on the given
+      deployment platform and no more.
+
+    It is possible to write high-level MirageOS applications, such as HTTPS,
+    email or CalDAV servers which can be deployed on very heterogenous and
+    embedded platforms by changing only a few compilation parameters. The
+    supported platforms range from minimal virtual machines running on cloud
+    providers, or processes running inside Docker containers configured with a
+    tight security profile. In general, these platform do not have a full POSIX
+    environment; MirageOS does not try to emulate POSIX and focuses on providing
+    a small, well-defined, typed interface with the system components. The
+    nearest equivalent to the MirageOS approach is the WASI (wasi.dev) set of
+    interfaces for WebAssembly.
+
+    {2 Is everything really written in OCaml?}
+
+    While most of the code is written in OCaml, a typed, high-level language
+    with many good safety properties, there are pieces of MirageOS which are
+    still written in C. These bits can be separated in three categories:
+
+    - The OCaml runtime is written in C. It needs to be ported to the platform
+      that MirageOS is trying to target, which do not support POSIX. Hence, the
+      first component to port to a new platform is the OCaml runtime.
+
+    - The low-level device drivers (network, console, clock, etc) also need some
+      C bits.
+
+    - The base usual C bindings; some libraries are widely used and
+      (unfortunately) very hard (but not impossible) to replace them completely
+      without taking a big performance hit or having to trust code without much
+      real-world usages. This is the case for low-level bit handling for crypto
+      code (even if we try to make sure allocation is alway handled by the OCaml
+      runtime) as well as arbitrary precision numeric computation (e.g. gmp).
+      Ideally we could image rewriting all of these libraries in OCaml if we had
+      an infinite amount of time in our hands.
+
+    {2 MirageOS as a cross-compilator}
+
+    The MirageOS compiler is basically a cross-compiler, where the host and
+    target toolchain are identical, but with different flags for the C bindings:
```

Comment:
maybe refer to the used pkg-config here? i.e. that CFLAGS are passed from the lowest layers via pkg-config files and invocation.

Reply:
I think a mention of `pkg-config` and `ocamlfind` as tools needed to correctly handle the cross-compilation is good.

Reply:
I've tried to complete that section with more information about pkg-config and ocamlfind predicates.

Reply:
Seems good for me :+1:.